### PR TITLE
QE: Remove/replace traditional client tests

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

**Related card**: https://github.com/SUSE/spacewalk/issues/17914

We already removed the deployment of a traditional client in our Uyuni CI Test environment.
Let's remove/replace traditional client tests.

**Warning**: To be merged only when

- [x] we give the ok for SUMA 4.3.1 submissions
- [x] we create the branch of Manager-4.3 in our SUSE private repository.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- Documentation needed: We removed documentation related to traditional clients

- [x] **DONE**

## Test coverage
- Cucumber tests were changed

- [x] **DONE**

## Links

NO PORTS
fixes https://github.com/SUSE/spacewalk/issues/13701

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by srbarrios** [comment]
Testing it here: https://ci.suse.de/blue/organizations/jenkins/uyuni-prs-ci-tests/detail/uyuni-prs-ci-tests/1522/pipeline

File: testsuite/features/step_definitions/common_steps.rb
Comment (nodeg) [inline testsuite/features/step_definitions/common_steps.rb]:
```suggestion
When(/^I wait until onboarding is completed for "([^"]*)"((?: salt minion)?)$/) do |host|
```
`is_salt` is not used anymore.

File: testsuite/features/build_validation/smoke_tests/smoke_tests.template
Comment (vandabarata) [inline testsuite/features/build_validation/smoke_tests/smoke_tests.template:46]: since, in the BV testsuite, it looks like client is being used generically, do we need to remove this particular line as well? or was it only necessary for traditional clients? I assume this because nothing else got changed for the BV "<client>"

File: testsuite/features/secondary/allcli_action_chain.feature
Comment (nodeg) [inline testsuite/features/secondary/allcli_action_chain.feature:4]: I wonder why we replace the SLE client with the CentOS minion in this feature. I guess because we already have a SLE minion and did the scenarios for the SLE client. I am wonderning if we need the cuplicate testing for a Salt minion here?

File: testsuite/features/step_definitions/common_steps.rb
Comment (nodeg) [inline testsuite/features/step_definitions/common_steps.rb:682]: Shouldn't we remove the word `client` here since we only reference it with the traditional client?

File: testsuite/features/secondary/min_ssh_tunnel.feature
Comment (nodeg) [inline testsuite/features/secondary/min_ssh_tunnel.feature:47]: Are you sure about removing this? Same with the SSH minion below.

File: testsuite/features/secondary/srv_maintenance_windows.feature
Comment (nodeg) [inline testsuite/features/secondary/srv_maintenance_windows.feature:5]: Nice catch!

File: testsuite/features/step_definitions/common_steps.rb
Comment (nodeg) [inline testsuite/features/step_definitions/common_steps.rb:181]: This was for the client? Do we need if for the minion, too? If yes we should update the name of the step.

File: testsuite/features/support/commonlib.rb
Comment (nodeg) [inline testsuite/features/support/commonlib.rb:178]: I wonder where this method is used throughout our code. Did you had a look because I did not find anything in this PR regarding replacing it.

**Review by nodeg — COMMENTED** [review]
I left some comments.

**Review by Bischoff — APPROVED** [review]
please wait for the merge of the renaming PR

File: testsuite/features/support/commonlib.rb
Comment (Bischoff) [reply testsuite/features/support/commonlib.rb:178]: it's used there:
```
testsuite/features/step_definitions/command_steps.rb:  if get_client_type(host) == 'traditional'
testsuite/features/step_definitions/common_steps.rb:  if get_client_type(host) == 'traditional' and is_salt.empty?
```

File: testsuite/features/secondary/min_ssh_tunnel.feature
Comment (Bischoff) [reply testsuite/features/secondary/min_ssh_tunnel.feature:47]: Yes, rhn_check does not exist on minions, it's trad only

File: testsuite/features/step_definitions/common_steps.rb
Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb:682]: +1 "minion client", while correct, sounds weird

File: testsuite/features/secondary/allcli_action_chain.feature
Comment (srbarrios) [reply testsuite/features/secondary/allcli_action_chain.feature:4]: Those features starting with allcli_ are features using multiple systems. What I did is to replace in some cases the sle_client by a centos_minion because both are using rpm files.
As the other option we have are the ubuntu_minion using deb files. And ofc, the sle_minion which is already in use in this feature.

File: testsuite/features/step_definitions/common_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/common_steps.rb:682]: Ok, I can do this refactor too

File: testsuite/features/secondary/min_ssh_tunnel.feature
Comment (srbarrios) [reply testsuite/features/secondary/min_ssh_tunnel.feature:47]: Yup

File: testsuite/features/step_definitions/common_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/common_steps.rb:181]: In theory, SUMA clients are all the clients we register in SUMA.
We can use the word client, the word system, and from now on, we can also use the word minion too :)

But what I will do is clarify this by passing by parameter the host we want to use.

File: testsuite/features/support/commonlib.rb
Comment (srbarrios) [reply testsuite/features/support/commonlib.rb:178]: Which was also removed. So all good.

File: testsuite/features/step_definitions/common_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/common_steps.rb]: Yeah, that was already fixed when I saw it in Rubocop.

**Comment by srbarrios** [comment]
PR testsuite is failing, and I can't try it yet.
Let's wait before merging it.

**Comment by nodeg** [comment]
The part about `spacewalk-channel` in `testsuite/documentation/cucumber-steps.md` can also be removed.
See https://github.com/SUSE/spacewalk/issues/13701

**Comment by srbarrios** [comment]
Testing the branch here 
https://ci.suse.de/blue/organizations/jenkins/uyuni-prs-ci-tests/detail/uyuni-prs-ci-tests/1654/pipeline

**Pull request merged**
